File: 0682-baseball-game/0682-baseball-game.py
class Solution:
    def calPoints(self, operations: list[str]) -> int:
        score_stack = deque()

        for operation in operations:
            match operation:
                case "+":
                    score_stack.append(score_stack[-1] + score_stack[-2])
                case "D":
                    score_stack.append(2 * score_stack[-1])
                case "C":
                    score_stack.pop()
                case _:
                    # Assume it's a numeric string
                    score_stack.append(int(operation))

        return sum(score_stack)

File: 0682-baseball-game/0682-baseball-game.go
func calPoints(operations []string) int {
    
    stack := []int{}

    for _, operation := range operations {
        switch operation {
            case "+":
                stack = append(stack, stack[len(stack) - 1] + stack[len(stack) - 2])

            case "D":
                stack = append(stack, stack[len(stack) - 1] * 2)

            case "C":
                stack = stack[:len(stack) - 1]

            default:
                value, _ := strconv.Atoi(operation)
                stack = append(stack, value)
        }
    }

    total := 0

    for _, value := range stack {
        total += value
    }

    return total
}